### PR TITLE
Fuzzer for the text parser.

### PR DESCRIPTION
**Pull request by msiebuhr**
No crashes detected after ~10M iterations.

(Stopped so I'd have battery enough left to submit this :))

### CODE REVIEW

File: text/fuzz/corpus/from_test_parse_error_0 [hunk 1]
```diff
@@ -0,0 +1 @@
+bla 3.14
```
Comment (juliusv) [inline text/fuzz/corpus/from_test_parse_error_0]: Please end any text files with a single newline character (http://stackoverflow.com/questions/729692/why-should-files-end-with-a-newline)

(or does that go against the purpose of the corpus items?)

File: text/fuzz/corpus/from_test_parse_1 [hunk 2]
```diff
@@ -0,0 +1,6 @@
+
```
Comment (juliusv) [inline text/fuzz/corpus/from_test_parse_1]: Wondering, is it intentional that all these corpus files start with a single empty line?

File: text/fuzz.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package text
+
+import "bytes"
+
+// Build only when actually fuzzing
+// +build gofuzz
```
Comment (juliusv) [inline text/fuzz.go]: Hmm, can we move this right after the license comment?

From the Go specs: "Constraints may appear in any kind of source file (not just Go), but they must appear near the top of the file, preceded only by blank lines and other line comments. These rules mean that in Go files a build constraint must appear before the package clause."

**Comment by juliusv** [comment]
Thanks for doing this! And cool to know as well that nothing has been found yet. The text format parser does get a lot more crazy input and throughput than the query language parser in world-wide day-to-day use, so maybe that already helped make it really stable (and of course @beorn7's diligence :)).

File: text/fuzz/corpus/from_test_parse_error_0 (hunk 1 above)
Comment (msiebuhr) [inline text/fuzz/corpus/from_test_parse_error_0]: All the `from_test_parse_` are written out as a one-time job from the test-suite, so it should be a clone of whatever was in there.

Comment (fabxc) [inline text/fuzz/corpus/from_test_parse_error_0]: Thanks a bunch for the effort!
Great work @beorn7!

**Comment by fabxc** [comment]
Thanks a bunch for the effort!
Great work @beorn7!

File: text/fuzz/corpus/from_test_parse_error_0 (hunk 1 above)
Comment (juliusv) [inline text/fuzz/corpus/from_test_parse_error_0]: Ah ok, of course. Cool then :)

**Comment by beorn7** [comment]
@juliusv @fabxc Is that good to merge?

@msiebuhr Thanks for the tests. And glad my code wasn't impressed... ;)

**Comment by beorn7** [comment]
I'm running this myself, at iteration 120M now.

**Comment by juliusv** [comment]
Yup, merging. Thanks! :+1:

**Pull request merged**